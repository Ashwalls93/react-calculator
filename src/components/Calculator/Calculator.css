.calculator-container {
  height: var(--calculator-height);
  width: var(--calculator-width);
  transition: var(--default-transition);
}

.calculator-theme-default {
  background-color: var(--calculator-background-color-default);
}

.calculator-theme-pony {
  background-color: var(--calculator-background-color-pony);
  border-radius: 8px;
}

.calculator-theme-dondi {
  background: url('../../assets/dondi.jpg');
  border-radius: 8px;
}