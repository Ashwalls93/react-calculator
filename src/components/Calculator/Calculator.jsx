import React from 'react';
import Display from '../Display/Display';
import Keypad from '../Keypad/Keypad'

const Calculator = () => {
  return (
    <div className="calculator-container">
      <Display displayValue="" />
      <Keypad 
        numbers={[]}
        operators={[]}
      />
    </div>
  );
}

export default Calculator;