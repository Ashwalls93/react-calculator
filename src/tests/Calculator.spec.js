import React from 'react';
import {shallow} from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Calculator from '../components/Calculator/Calculator';

describe('Calculator', () => {
  it('should render', () => {
    const wrapper = shallow(<Calculator />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should contain the Display and Keypad Components', () => {
    const wrapper = shallow(<Calculator />);
    expect(wrapper.containsAllMatchingElements([
      <Display />,
      <Keypad />
    ])).to.equal(true);
  });
});



