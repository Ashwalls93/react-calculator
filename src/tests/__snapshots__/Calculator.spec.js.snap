// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator should render 1`] = `
<div
  className="calculator-container"
/>
`;
